Shopify/tokengating-example-app
# :pushpin: Update dependencies for packages

## Related issues

https://github.com/Shopify/blockchain-components/issues/25
https://github.com/Shopify/tokengating-example-app/issues/3

## Context

Updates outdated packages and removes strict node engine enforcement. I've added comments to individual files which have changed for more granular context as well.

Additionally,
1. Adds `package-lock.json` to gitignore
2. Updates packages to most recent versions
3. Addresses some import issues for connect-wallet inside of `tokengate-src`
4. Changes vite polyfill package to one that supports v4+


## Demonstration

<details><summary>Demonstration shows a clean pull and running install which shows no dependency conflicts.</summary>
<p>

https://user-images.githubusercontent.com/4250423/220971758-60a08892-cc9d-4155-b119-abf7e33acd13.mov

</p>
</details> 

## How to test

- Pull this branch
- Run `npm install`
- Build the packages + deploy
- The packages should function as expected
